include/effects/matrix/PatternSMSupernova.h

```diff
@@ -137,17 +135,23 @@ class PatternSMSupernova : public LEDStripEffect
 
     void drawPixelXYF(float x, float y, CRGB color)
     {
+        #if SHOW_VU_METER
+        static constexpr bool showingVUMeter = true;
+        #else
+        static constexpr bool showingVUMeter = false;
+        #endif
+
         const uint8_t xx = (x - (int)x) * 255, yy = (y - (int)y) * 255, ix = 255 - xx, iy = 255 - yy;
         const uint8_t wu[4] = {WU_WEIGHT(ix, iy), WU_WEIGHT(xx, iy), WU_WEIGHT(ix, yy), WU_WEIGHT(xx, yy)};
         for (uint8_t i = 0; i < 4; i++) {
-            // Until now, x and y are totally in bound, but we're about
-            // to make our coords into 4-pixel wide splotches.
-            // If we knew x and y would be ^2, we could do cute bit math,
-            // but we just clamp instead.
-            const int xn = std::clamp((int)x + (i & 1), 0, MATRIX_WIDTH - 1);
-            const int yn = std::clamp((int)y + ((i >> 1) & 1), 0, MATRIX_HEIGHT - 1);
-
-            CRGB clr = g()->leds[XY(xn, MATRIX_HEIGHT - yn)];
+            const int xn = x + (i & 1);
+            const int yn = y + ((i >> 1) & 1);
+
+            // Make sure we're on the panel and leave the VU meter pixels alone, if we're showing it
+            if (!g()->isValidPixel(xn, yn) || (showingVUMeter && yn == (MATRIX_HEIGHT - 1)))
```

Comment:
Why the extra level of labels?  Why not (SHOW_VU_METER) directly in the expression, since it will always be defined?

Reply:
It actually isn't always defined. It's only defined for matrix/HUB75 configurations, and even then not for all - env:mesmerizer_lolin doesn't, as I found when CI ran on an earlier commit. However, @robertlipe made a good alternative suggestion, which I will implement.